src/class/video/video_device.c

```diff
@@ -251,39 +250,39 @@ static void const* _find_desc_ep(void const *beg, void const *end)
  *
  * @return The pointer for interface descriptor.
  * @retval end   did not found interface descriptor */
-static void const* _find_desc_entity(void const *desc, uint_fast8_t entityid)
+static uint8_t const* _find_desc_entity(uint8_t const *desc, uint_fast8_t entityid)
 {
   tusb_desc_vc_itf_t const *vc = (tusb_desc_vc_itf_t const*)desc;
-  void const *beg = vc;
-  void const *end = beg + vc->std.bLength + vc->ctl.wTotalLength;
-  for (void const *cur = beg; cur < end; cur = _find_desc(cur, end, TUSB_DESC_CS_INTERFACE)) {
+  uint8_t const *beg = (uint8_t const *) vc;
+  uint8_t const *end = beg + vc->std.bLength + vc->ctl.wTotalLength;
+  for (uint8_t const *cur = beg; cur < end; cur = _find_desc(cur, end, TUSB_DESC_CS_INTERFACE)) {
     tusb_desc_cs_video_entity_itf_t const *itf = (tusb_desc_cs_video_entity_itf_t const *)cur;
     if ((VIDEO_CS_ITF_VC_INPUT_TERMINAL  <= itf->bDescriptorSubtype
          && itf->bDescriptorSubtype < VIDEO_CS_ITF_VC_MAX)
         && itf->bEntityId == entityid) {
-      return itf;
+      return cur;
     }
     cur = tu_desc_next(cur);
   }
   return end;
 }
 
 /** Return the end of the video streaming descriptor. */
-static inline void const* _end_of_streaming_descriptor(void const *desc)
+static inline uint8_t const* _end_of_streaming_descriptor(uint8_t const *desc)
```

Comment:
The modification makes caller require cast operations. Does it need for IAR?

I think it is better to avoid cast operations at the caller. Cast operations sometimes occur unexpected bugs, it is better to gather cast operations in a low-level function than scatter at caller functions.

Reply:
It was just the case that there was a lot of pointer arithmetic on void pointers, and it was easiest to simply replace all of them in that source file with `uint8_t const *`. I have gone through and reverted them to `void const *` where that reduces the overall number of casts involved. Overall, the commit now results in a net reduction in the number of casts, which is nice.

Reply:
Thank you for your comments and updating!